test_conformance/api/test_queue_array_properties.cpp

```diff
@@ -0,0 +1,294 @@
+        test_error(error, "clCreateCommandQueueWithProperties failed");
+    }
+    std::vector<cl_queue_properties> get_properties;
+    size_t set_size = 0;
+
+    error = clGetCommandQueueInfo(test_queue, CL_QUEUE_PROPERTIES_ARRAY, 0,
+                                  NULL, &set_size);
+    test_error(
+        error,
+        "clGetCommandQueueInfo failed asking for CL_QUEUE_PROPERTIES_ARRAY");
+
+    if (test_case.properties.size() == 0 && set_size == 0)
+    {
+        return TEST_PASS;
+    }
+
+    cl_uint number_of_props = set_size / sizeof(cl_queue_properties);
+    get_properties.resize(number_of_props);
+    error = clGetCommandQueueInfo(test_queue, CL_QUEUE_PROPERTIES_ARRAY,
+                                  set_size, get_properties.data(), NULL);
+    test_error(
+        error,
+        "clGetCommandQueueInfo failed asking for CL_QUEUE_PROPERTIES_ARRAY");
+
+    if (get_properties.back() != 0)
+    {
+        log_error("ERROR: Incorrect last property value - should be 0!\n");
+        return TEST_FAIL;
+    }
+    get_properties.pop_back();
+    test_case.properties.pop_back();
+
+    if (get_properties != test_case.properties)
+    {
+        for (cl_uint i = 0; i < test_case.properties.size(); i = i + 2)
+        {
+            cl_queue_properties set_property = test_case.properties[i];
+            cl_queue_properties set_property_value =
+                test_case.properties[i + 1];
+
+            std::vector<cl_mem_properties>::iterator it = std::find(
+                get_properties.begin(), get_properties.end(), set_property);
+
+            if (it == get_properties.end())
+            {
+                log_error("ERROR: Property not found ... 0x%x\n", set_property);
+                return TEST_FAIL;
+            }
+            else
+            {
+                if (set_property_value != *std::next(it))
+                {
+                    log_error("ERROR: Incorrect preperty value expected %x, "
+                              "obtained %x\n",
+                              set_property_value, *std::next(it));
+                    return TEST_FAIL;
+                }
+            }
+        }
+        log_error("ERROR: ALL properties and values matched but order "
+                  "incorrect!\n");
+        return TEST_FAIL;
+    }
+    return error;
+}
+
+int run_test_queue_array_properties(cl_context context, cl_device_id deviceID,
+                                    test_queue_array_properties_data test_case)
+{
+    int error = CL_SUCCESS;
+    clCommandQueueWrapper queue;
+    std::vector<cl_queue_properties> requested_properties =
+        test_case.properties;
+    log_info("TC description: %s\n", test_case.description.c_str());
+    // first verify if user properties are supported
+    if (requested_properties.size() != 0)
+    {
+        requested_properties.pop_back();
+        cl_command_queue_properties requested_bitfield;
```

Comment:
should we initialize this to zero in case there's a `CL_QUEUE_SIZE` property but no `CL_QUEUE_PROPERTIES`? could add a test case for that too